react/ProductHighlights.tsx

```diff
@@ -1,26 +1,43 @@
-import React, { FC, useContext, ReactNode, useMemo } from 'react'
-import { useProduct, ProductTypes } from 'vtex.product-context'
+import type { FC, ReactNode } from 'react'
+import React, { useContext, useMemo } from 'react'
+import { defineMessages } from 'react-intl'
+import type { ProductTypes } from 'vtex.product-context'
+import { useProduct } from 'vtex.product-context'
 
+import { FILTER_TYPE, QUERY_TYPE } from './constants/EnumTypes'
 import { getSeller } from './modules/seller'
 
-type HighlightType = 'collection' | 'promotion' | 'teaser'
+type HighlightType =
+  | typeof QUERY_TYPE.collection
+  | typeof QUERY_TYPE.promotion
+  | typeof QUERY_TYPE.teaser
+
+interface Filter {
+  type: typeof FILTER_TYPE.hide | typeof FILTER_TYPE.show
```

Comment:
```suggestion
  type: 'hide' | 'show'
```

Reply:
This type has the same problem I pointed out in the previous comment.
`Filter.type` would accept any `string`.

Reply:
@victorhmp

Correction made.